src/dnssec/sign/keys/keyset.rs

```diff
@@ -149,6 +199,10 @@ impl KeySet {
         }
     }
 
+    fn unique_key_tag(&self, key_tag: u16) -> bool {
+        !self.keys.iter().any(|(_, k)| k.key_tag == key_tag)
```

Comment:
Nit: a minor change would better match (IMO) the fn name:
```suggestion
        self.keys.iter().all(|(_, k)| k.key_tag != key_tag)
```